### PR TITLE
DNM  [nrf mergeup] merge upstream up to commit ebd0503

### PR DESCRIPTION
**Pull request by nvlsianpu**
* Replace flash partitioning terminology
* Change ec256 interface so that it could support multiple interfaces in
the future.
* zephyr bugfix: mcuboot doesn't link into code-partition
* image encryption: fix enc_state array indexing for zephyr
* Fix snprintf warning 
* DFU over USB class (so fare works on intel xtensa S1000 board)

### CODE REVIEW

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -43,6 +44,29 @@ config BOOT_USE_TINYCRYPT
...
+	# will collide.
+	depends on ! MBEDTLS
+	help
+	  Use cc310 for crypto primitives.
+
+config BOOT_USE_NRF_CC310_BL
+	bool
+	default n
+
+config NRFXLIB_CRYPTO
```
Comment (sigvartmh) [inline boot/zephyr/Kconfig]: I don't think this is needed for downstream as we get the Kconfig from nrfxlib? through the west manifest in nrf

File: boot/zephyr/Kconfig [hunk 2]
```diff
@@ -43,6 +44,29 @@ config BOOT_USE_TINYCRYPT
...
+	bool
+	default n
+
+config NRF_CC310_BL
```
Comment (sigvartmh) [inline boot/zephyr/Kconfig]: I think this is redundant I'll fix it upstream as it's defined sort of twice in L93-94

File: boot/zephyr/Kconfig (hunk 1 above)
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: @SebastianBoe ^^

File: boot/zephyr/Kconfig (hunk 2 above)
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: ok

File: boot/zephyr/Kconfig (hunk 1 above)
Comment (sigvartmh) [reply boot/zephyr/Kconfig]: I'll remove it and add a west manifest to MCUBoot upstream

Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: can you do it quickly here as [nrf fromlist] commit and send path to upsterem as well?

File: boot/zephyr/Kconfig (hunk 2 above)
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: can you do it quickly here as [nrf fromlist] commit and send path to upsterem as well?

File: boot/zephyr/Kconfig (hunk 1 above)
Comment (sigvartmh) [reply boot/zephyr/Kconfig]: Sure I'll do that

**Comment by nvlsianpu** [comment]
@sigvartmh I've updated missing signed-of-by lines.

File: boot/zephyr/Kconfig (hunk 2 above)
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: This is still in two places NCS/nrfxlib/crypto/Kconfig and NCS/mcuboot/boot/zephyr/Kconfig

Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: @sigvartmh  ^^

Comment (sigvartmh) [reply boot/zephyr/Kconfig]: It should be in  NCS/nrfxlib/crypto/Kconfig but not in  NCS/mcuboot/boot/zephyr/Kconfig I thought I removed it.

Comment (sigvartmh) [reply boot/zephyr/Kconfig]: Do you want me to fix it maybe? then make a PR towards your branch with changes to the remove duplicate commit?

Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: @sigvartmh  fix on my branch, I'll give you permissions.

Comment (sigvartmh) [reply boot/zephyr/Kconfig]: okei :+1:

Comment (sigvartmh) [reply boot/zephyr/Kconfig]: Fixed @nvlsianpu

**Comment by nvlsianpu** [comment]
@carlescufi Can you look at last 3 commits.

**Comment by nvlsianpu** [comment]
@sigvartmh 
> Can you explain why we need a manifest in MCUboot?

I think it is not needed. Why to add it? Build system is working without this as well in stand-alone fashion.

**Comment by sigvartmh** [comment]
> @sigvartmh
> 
> > Can you explain why we need a manifest in MCUboot?
> 
> I think it is not needed. Why to add it? Build system is working without this as well in stand-alone fashion.

It's to support the use of  `NRF_CC310`(Hardware crypto acceleration) in MCUBoot. If you get zephyr-project/zephyr the manifest file in zephyr does not include the nrfxlib repository. Hence you'll get a build error if you try to build MCUBoot with CC310 enabled. The manifest is mainly there to allow the use of `nrfx_crypto` without binding MCUBoot against ncs/zephyr and ncs/nrf.

I do see that this does not really make sense downstream but upstream this makes sense todo as MCUBoot don't want to be dependent on ncs/nrf.

@carlescufi @nvlsianpu: Not sure if this gave enough reasoning behind it and that the reasoning makes sense.

**Comment by nvlsianpu** [comment]
@sigvartmh  Manifest in mcuboot doesn't make sense as we suppose to build NCS mcuboot using NCS component.
I removed commit with the manifest.

**Comment by sigvartmh** [comment]
That's fine by me :)

**Comment by nvlsianpu** [comment]
@carlescufi gitlint faliure. I fixed on our commits, but it fails due the merged commit as well - have to ignore it then.

**Pull request merged**